x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/Count.java

```diff
@@ -125,8 +131,10 @@ public Expression surrogate() {
                 new Coalesce(s, new MvCount(s, field), List.of(new Literal(s, 0, DataType.INTEGER))),
                 new Count(s, new Literal(s, StringUtils.WILDCARD, DataType.KEYWORD))
             );
+        } else if (field instanceof AggregateDoubleMetricAttribute aggregateDoubleMetricAttribute) {
+            return new Count(source(), aggregateDoubleMetricAttribute.getValueCountSubField());
```

Comment:
I think this should be `Sum(source(), aggregateDoubleMetricAttribute.getValueCountSubField())` and we can remove the override `supplier` above?